File: app/src/main/java/com/softwareverde/tidyduck/most/Company.java
package com.softwareverde.tidyduck.most;

import com.softwareverde.json.Json;
import com.softwareverde.json.Jsonable;

public class Company implements Jsonable {
    private Long _id;
    private String _name;

    public Long getId() {
        return this._id;
    }

    public void setId(long id) {
        this._id = id;
    }

    public String getName() {
        return this._name;
    }

    public void setName(String name) {
        this._name = name;
    }

    @Override
    public Json toJson() {
        final Json json = new Json(false);

        json.put("id", _id);
        json.put("name", _name);

        return json;
    }
}

File: app/src/main/webapp/js/classes/company.js
class Company {
    static fromJson(json) {
        if (json == null) {
            return null;
        }
        const company = new Company();

        company.setId(json.id);
        company.setName(json.name);

        return company;
    }

    static toJson(company) {
        return {
            id: company.getId(),
            name: company.getName()
        };
    }

    constructor() {
        this._id = null;
        this._name  = null;
    }

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setName(name) {
        this._name = name;
    }

    getName() {
        return this._name;
    }
}
